WordPress/gutenberg
# Site Editor: Fix deprecation console error in top toolbar

## What?
This PR fixes a browser warning error that is logged when opening the editor canvas in the Site Editor.

```
LFJDOMBA.js:48 The `as` prop is deprecated. Use the `render` prop instead. See https://ariakit.org/guide/composition
```

https://github.com/WordPress/gutenberg/assets/54422211/931b3091-1bf3-49d7-8e18-79e29ca8d134

## Why?

According to my research, I found that this error occurred in the `HeaderEditMode` component. Inside this component, the `as` property is used on the `NavigableToolbar`. And this component is internally rendering `Ariakit.Toolbar`. As the console message indicates, I expected that we should use the `render` property instead of the `as` property.

## How?

~I simply changed the `as` property to the `render` property.~ Wrap `NavigableToolbar` in a `motion.div`.

## Testing Instructions

- Open the Site Editor.
- Click the editor canvas.
- You should not see any deprecation errors in your browser console.
- Enable Distraction free mode.
- The buttons on the left side of the toolbar should animate when you hover over the top of the canvas.

## Screenshots or screencast 

https://github.com/WordPress/gutenberg/assets/54422211/f0be08c0-6883-456f-bb09-6f7c351731cb
